src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -51,7 +51,11 @@ public function __construct(ContainerInterface $container, $resource, array $opt
     public function getRouteCollection()
     {
         if (null === $this->collection) {
+            $this->collection = new RouteCollection();
```

Comment:
It's not the responsibility of the router class to do that. It's the responsibility of the calling code to ensure it does not call `getRouteCollection` twice when the first call throws an exception. This is obvious when you consider that setting an empty RouteCollection does not make sense semantically.